depthai_nodes/parsing_neural_network.py

```diff
@@ -0,0 +1,130 @@
+from pathlib import Path
+
+import depthai as dai
+
+from .ml.parsers import BaseParser, ParserGenerator
+
+
+class ParsingNeuralNetwork(dai.node.ThreadedHostNode):
```

Comment:
Does this need to be defined as a host node? I assume it's not possible to just extend the NeuralNetwork node, right?

Reply:
Yea, I tried to inherit from NeuralNetwork, but DepthAI doesn't allow that. I wouldn't view this in bad way thought, because a lot of times it's better to favor composition over inheritance.